pennylane/tape/tapes/tape.py

```diff
@@ -330,6 +332,15 @@ def _process_queue(self):
             elif isinstance(obj, qml.operation.Observable) and "owner" not in info:
                 raise ValueError(f"Observable {obj} does not have a measurement type specified.")
 
+        m_wires = []  # The wires upon which an observable is measured
+        for m in self._measurements:
+            m_wires.extend([w for w in m.wires if m.return_type in OBSERVABLE_MEASUREMENTS])
+
+        if len(m_wires) != len(set(m_wires)):
+            raise qml.QuantumFunctionError(
+                "Each wire in the quantum circuit can only be measured by a single observable"
+            )
```

Comment:
Hey Tom, I just had a thought; we should probably instead check for `m.obs.wires`, if `obs` is not None. Only `Expectation`, `Variance`, and `Sample` `MeasurementProcess` objects have a non-None `obs` attribute, so this is a good way of _directly_ testing for observables on the same wire.